File: Data Structures & Algorithms/largest-rectangle-in-histogram/submission-5.cpp
class Solution {
public:
    int largestRectangleArea(vector<int>& heights) {
        int maxArea=INT_MIN;
        for(int i=0;i<heights.size();i++){
            int length=heights[i];
            int breadth=1;
            int left=i-1;
            int right=i+1;
            while(left >= 0 && heights[left] >=heights[i]){
                breadth++;
                left--;
            }
            while(right < heights.size() && heights[right] >=heights[i]){
                breadth++;
                right++;
            }
            int area=length * breadth;
            maxArea=max(maxArea,area);
        }
        return maxArea;
    }
};

File: Data Structures & Algorithms/max-water-container/submission-5.java
class Solution {
    public int maxArea(int[] heights) {
        int left=0;
        int right=heights.length-1;
        int maxArea=Integer.MIN_VALUE;
        while(left < right){
            int length=Math.min(heights[left],heights[right]);
            int breadth=right-left;
            int area=length*breadth;
            maxArea=Math.max(area,maxArea);
            if(heights[left]<heights[right]){
                left++;
            }
            else{
                right--;
            }
        }
        return maxArea;
    }
}
